src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,74 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final int INDEX_OF_NAME = 1;
+    private static final int INDEX_OF_WORKING_HOURS = 2;
+    private static final int INDEX_OF_INCOME_PER_HOUR = 3;
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+    private static final DateTimeFormatter
+            DATE_TIME_FORMAT = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final String DELIMITER_SPACE = " ";
+
+    private LocalDate convertStringToLocalDate(String date) {
+        return LocalDate.parse(date, DATE_TIME_FORMAT);
+    }
+
+    private String getName(String[] data, int index) {
+        String[] splitData = data[index].split(DELIMITER_SPACE);
+        return splitData[INDEX_OF_NAME];
+    }
+
+    private int getMoney(String[] data, int index) {
+        String[] splitData = data[index].split(DELIMITER_SPACE);
+        return Integer.parseInt(splitData[INDEX_OF_WORKING_HOURS])
+                * Integer.parseInt(splitData[INDEX_OF_INCOME_PER_HOUR]);
+    }
+
+    private boolean checkData(LocalDate date, LocalDate dateBeginning, LocalDate dateEnding) {
+        return !date.isBefore(dateBeginning) && !date.isAfter(dateEnding);
+    }
+
+    private void setMoney(String[] data, int index, int[] salary, String[] names) {
+        String name = getName(data, index);
+        for (int i = 0; i < names.length; ++i) {
+            if (names[i].equals(name)) {
+                salary[i] += getMoney(data, index);
+                return;
+            }
+        }
+    }
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
```

Comment:
public methods should be declared before private ones. 
with second priority we order methods by order of their call. 
correct order will be:

1. getSalaryInfo
2. countSalary
3. convertStringToLocalDate
4. checkData
5. setMoney
6. getMoney
7. getName

make related code changes and think about why my suggestion is as it is